src/components/QueryRecordings/QueryRecordings.vue

```diff
@@ -115,246 +94,168 @@ export default {
+      this.tagData = {
+        tagMode: "any"
+      };
+    },
     saveLastQuery() {
-      this.lastQuery = JSON.parse(JSON.stringify(this.query));
+      this.lastQuery = this.serialiseQueryForRecall();
+      this.$emit("description", this.makeSearchDescription());
+    },
+    queryHasChanged() {
+      return (
+        JSON.stringify(this.lastQuery) !==
+        JSON.stringify(this.serialiseQueryForRecall())
+      );
     },
     updatePagination(perPage, page) {
       this.query.limit = perPage;
       const newOffset = Math.max(0, (page - 1) * perPage);
       this.query.offset = newOffset;
       this.updateRouteQuery();
+      this.makeApiRequest();
     },
-    deserialiseRouteIntoQuery(routeQuery, target) {
-      target = target || this.query;
-      for (const key in routeQuery) {
-        if (routeQuery.hasOwnProperty(key)) {
-          target[key] = routeQuery[key];
-        }
+    setOnlyIfExists(itemName, source, destination) {
+      if (source.hasOwnProperty(itemName)) {
+        destination[itemName] = source[itemName];
       }
-      if (routeQuery.where) {
-        target.where = JSON.parse(routeQuery.where);
-        if (!target.where.recordingDateTime) {
-          this.$set(target.where, "recordingDateTime", {});
-        }
-        if (!target.where.dateRange) {
-          this.$set(target.where, "dateRange", {});
-        } else {
-          if (target.where.dateRange === "customDateRange") {
-            this.$set(target.where, "dateRange", { isCustom: true });
-          } else if (target.where.dateRange === "all") {
-            this.$set(target.where, "dateRange", { all: true });
-          }
-        }
-        if (!target.where.hasOwnProperty("DeviceId")) {
-          this.$set(target.where, "DeviceId", []);
-        }
-        if (!target.where.duration) {
-          this.$set(target.where, "duration", {});
-        }
-        if (target.where.DeviceId && target.where.DeviceGroups) {
-          target.where.DeviceId = [
-            ...target.where.DeviceId,
-            ...target.where.DeviceGroups
-          ];
-        } else if (target.where.DeviceGroups) {
-          target.where.DeviceId = [...target.where.DeviceGroups];
-        }
+    },
+    makeArray(value) {
+      if (typeof value === "object") {
+        return value;
+      } else {
+        return value ? [value] : [];
       }
-      if (routeQuery.tags) {
-        target.tags = JSON.parse(routeQuery.tags);
+    },
+    setAdvancedInitalState() {
+      // If there was an advanced query, start with the advanced toggle area open.
+      this.advanced =
+        this.hasTagData() ||
+        this.duration.hasOwnProperty("maxS") ||
+        this.duration.hasOwnProperty("minS");
+    },
+    hasTagData() {
+      return (
+        this.tagMode !== "any" ||
+        (this.tagData.tags && this.tagData.tags.length > 1)
+      );
+    },
+    deserialiseRouteIntoQuery(routeQuery) {
+      this.setOnlyIfExists("offset", routeQuery, this.query);
```

Comment:
Probably helper functions like `setOnlyIfExists` shouldn't be a method on the component; they don't need access to `this` since all the args are passed in.